modules/fastcv/test/test_warp.cpp

```diff
@@ -40,30 +29,80 @@ TEST_P(WarpPerspective2Plane, accuracy)
     {
         s[i].x += buffer[i*4]*src.cols/2;
         s[i].y += buffer[i*4+1]*src.rows/2;
-        d[i].x += buffer[i*4+2]*dst1.cols/2;
-        d[i].y += buffer[i*4+3]*dst1.rows/2;
+        d[i].x += buffer[i*4+2]*dst.cols/2;
+        d[i].y += buffer[i*4+3]*dst.rows/2;
     }
 
-    cv::getPerspectiveTransform( s, d ).convertTo( mat, mat.depth() );
+    cv::getPerspectiveTransform( s, d ).convertTo( M, M.depth() );
+
     // Invert the perspective matrix
-    invert(mat,mat);
+    invert(M,M);
+}
+
+typedef testing::TestWithParam<cv::Size> WarpPerspective2Plane;
+
+TEST_P(WarpPerspective2Plane, accuracy)
+{
+    cv::Size dstSize = GetParam();
+    cv::Mat src = imread(cvtest::findDataFile("cv/shared/baboon.png"), cv::IMREAD_GRAYSCALE);
+    cv::Mat dst1, dst2, matrix, ref1, ref2;
+    matrix.create(3, 3, CV_32FC1);
+    dst1.create(dstSize, CV_8UC1);
+    dst2.create(dstSize, CV_8UC1);
 
-    cv::fastcv::warpPerspective2Plane(src, src, dst1, dst2, mat, dstSize);
-    cv::warpPerspective(src,ref1,mat,dstSize,(cv::INTER_LINEAR | cv::WARP_INVERSE_MAP));
-    cv::warpPerspective(src,ref2,mat,dstSize,(cv::INTER_LINEAR | cv::WARP_INVERSE_MAP));
+    getInvertMatrix(src, dst1, matrix);
 
-    cv::Mat difference1, difference2, mask1,mask2;
+    cv::fastcv::warpPerspective2Plane(src, src, dst1, dst2, matrix, dstSize);
+    cv::warpPerspective(src, ref1, matrix, dstSize, (cv::INTER_LINEAR | cv::WARP_INVERSE_MAP),cv::BORDER_CONSTANT,Scalar(0));
+    cv::warpPerspective(src, ref2, matrix, dstSize, (cv::INTER_LINEAR | cv::WARP_INVERSE_MAP),cv::BORDER_CONSTANT,Scalar(0));
+
+    cv::Mat difference1, difference2, mask1, mask2;
     cv::absdiff(dst1, ref1, difference1);
     cv::absdiff(dst2, ref2, difference2);
+
+    // There are 1 or 2 difference in pixel value because algorithm is different, ignore those difference
     cv::threshold(difference1, mask1, 5, 255, cv::THRESH_BINARY);
     cv::threshold(difference2, mask2, 5, 255, cv::THRESH_BINARY);
     int num_diff_pixels_1 = cv::countNonZero(mask1);
     int num_diff_pixels_2 = cv::countNonZero(mask2);
 
-    EXPECT_LT(num_diff_pixels_1, src.size().area()*0.02);
-    EXPECT_LT(num_diff_pixels_2, src.size().area()*0.02);
+    // The border is different
+    EXPECT_LT(num_diff_pixels_1, (dstSize.width+dstSize.height)*5);
+    EXPECT_LT(num_diff_pixels_2, (dstSize.width+dstSize.height)*5);
+}
+
+typedef testing::TestWithParam<tuple<Size, int, int>> WarpPerspective;
+
+TEST_P(WarpPerspective, accuracy)
+{
+    cv::Size dstSize = get<0>(GetParam());
+    int interplation = get<1>(GetParam());
+    int borderType   = get<2>(GetParam());
+    cv::Scalar borderValue = Scalar::all(100);
+
+    cv::Mat src = imread(cvtest::findDataFile("cv/shared/baboon.png"), cv::IMREAD_GRAYSCALE);
+    cv::Mat dst, matrix, ref;
+    matrix.create(3, 3, CV_32FC1);
+    dst.create(dstSize, src.type());
```

Comment:
There is no sense to create output buffer before the call. The further function should do it for you.